packages/flutter/lib/src/material/material_localizations.dart

```diff
@@ -119,22 +131,45 @@ class DefaultMaterialLocalizations implements MaterialLocalizations {
   ///
   /// [LocalizationsDelegate] implementations typically call the static [load]
   /// function, rather than constructing this class directly.
-  DefaultMaterialLocalizations(this.locale) {
-    assert(locale != null);
-
+  DefaultMaterialLocalizations(this.locale)
+      : assert(locale != null),
+        this._localeName = _computeLocaleName(locale) {
     if (localizations.containsKey(locale.languageCode))
       _nameToValue.addAll(localizations[locale.languageCode]);
     if (localizations.containsKey(_localeName))
       _nameToValue.addAll(localizations[_localeName]);
+
+    if (intl.NumberFormat.localeExists(_localeName)) {
```

Comment:
@HansMuller @yjbanov We really need to fix https://github.com/flutter/flutter/issues/12050 before we add more `intl` dependencies.